engine/resources/symbolic_test.go

```diff
@@ -0,0 +1,42 @@
+package resources
+
+import (
+	"testing"
+)
+
+func Test_mode(t *testing.T) {
+	testCases := []struct {
+		input string
+		want  string
+		err   error
+	}{
+		{input: "ug=r,go=r,ua=x", want: "0555"},
+		{input: "uu=rw,ug=rw,gg=xr", want: "0670"},
+		{input: "u=r", want: "0400"},
+		{input: "g=rw", want: "0060"},
+		{input: "o=rwx", want: "0007"},
+		{input: "u=r", want: "0400"},
+		{input: "ugo=r,g=w,o=w", want: "0466"},
+		{input: "u=r,g=w,o=x", want: "0421"},
+		{input: "ug=rw,o=rx", want: "0665"},
+		{input: "g=r,ugo=x,a=w", want: "0373"},
+
+		// currently failing tests
+		// {input: "o=args", err: fmt.Errorf("permission setting failed")},
+		// {input: "o=arq", err: fmt.Errorf("permission setting failed")},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.input, func(t *testing.T) {
```

Comment:
There's a possible bug here... It's tricky. Learn it and explain it...

https://golang.org/doc/articles/race_detector.html#Race_on_loop_counter

Reply:
I ran this package's test with the race detector and couldn't find any race condition.
My guess would be `t *testing.T` potentially being an accidentally shared variable between `func(t *testing.T)` and `Test_mode(t *testing.T)`